service/payment.go

```diff
@@ -63,15 +63,22 @@ func (paymentStatus PaymentStatus) String() string {
 // CreatePaymentSession creates a payment session and returns a journey URL for the calling app to redirect to
 func (service *PaymentService) CreatePaymentSession(req *http.Request, createResource models.IncomingPaymentResourceRequest) (*models.PaymentResourceRest, ResponseType, error) {
 
+	err := validateIncomingPayment(createResource, &service.Config)
+	if err != nil {
+		err = fmt.Errorf("invalid incoming payment: [%v]", err)
+		log.ErrorR(req, err)
+		return nil, InvalidData, err
+	}
+
 	// Get user details from context, put there by UserAuthenticationInterceptor
 	userDetails, ok := req.Context().Value(helpers.ContextKeyUserDetails).(models.AuthUserDetails)
 	if !ok {
-		err := fmt.Errorf("invalid AuthUserDetails in request context")
+		err = fmt.Errorf("invalid AuthUserDetails in request context")
 		log.ErrorR(req, err)
 		return nil, InvalidData, err
```

Comment:
not actually part of this PR but while here and adding a case for invalid data ..... this error case should be a normal `Error` rather than `InvalidData` - the case for getting here means something is wrong in the routing within the service as it must have bypassed the user auth interceptor somehow - should not be possible hence generic internal server error

Reply:
Updated